compiler/rustc_interface/src/util.rs

```diff
@@ -482,27 +482,21 @@ fn categorize_crate_type(s: Symbol) -> Option<CrateType> {
 }
 
 pub fn collect_crate_types(session: &Session, attrs: &[ast::Attribute]) -> Vec<CrateType> {
-    // Unconditionally collect crate types from attributes to make them used
-    let attr_types: Vec<CrateType> = attrs
-        .iter()
-        .filter_map(|a| {
-            if a.has_name(sym::crate_type) {
-                match a.value_str() {
-                    Some(s) => categorize_crate_type(s),
-                    _ => None,
-                }
-            } else {
-                None
-            }
-        })
-        .collect();
-
     // If we're generating a test executable, then ignore all other output
     // styles at all other locations
     if session.opts.test {
         return vec![CrateType::Executable];
     }
 
+    // Unconditionally collect crate types from attributes to make them used
```

Comment:
"Unconditionally collect" part no longer true?

Reply:
TBH I didn't understand that comment, especially the "make them used" part.

Reply:
attributes need to be *marked as used*, otherwise a lint about an unused attribute gets emitted. Maybe this is not an issue with `crate_type` attributes, or we have no tests for `--test` executables with `crate_type` attributes

Reply:
Right, but this code doesn't seem relevant to that -- there's no state being recorded here. Maybe the comment is out of date?

Reply:
Oh! The exact same comment appears in `check_attr_crate_type`. So I think this comment is bogus and can be removed.

Reply:
yea that's odd... The same comment exists in the same file on a similar collection code. I think you're right and it's just a refactoring leftover.

I did some digging, and it used to be `check_name`, which indeed used to mark the attribute as used. I guess just remove the comment, it carries no more value.